File: src/main/java/com/github/kindrat/cassandra/client/ui/ConnectionData.java
package com.github.kindrat.cassandra.client.ui;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ConnectionData {
    /**
     * cassandra url containing host and port (localhost:9042)
     */
    private String url;
    /**
     * cassandra keyspace
     */
    private String keyspace;
    /**
     * cassandra username
     */
    private String username;
    /**
     * cassandra password
     */
    private String password;
}

File: src/main/kotlin/com/github/kindrat/cassandra/client/ui/ConnectionData.kt
package com.github.kindrat.cassandra.client.ui

import org.springframework.data.cassandra.config.CompressionType

data class ConnectionData(
        val url: String,
        val keyspace: String,
        val localDatacenter: String,
        val username: String?,
        val password: String?,
        val compressionType: CompressionType = CompressionType.LZ4
)